### PR TITLE
Add coherency between check and normal mode for copy plugin action see issue #24633

### LINKED ISSUE #24633: Missing information from registered var when template/copy change file contents in check mode
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
Modules: template, copy

##### ANSIBLE VERSION
Testing below from devel branch of git repo last pulled 5/15/2017
```
ansible 2.4.0 (devel 8dae2ea869) last updated 2017/05/15 15:38:53 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/szynaka/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/szynaka/src/ansible_upstream/lib/ansible
  executable location = /home/szynaka/src/ansible_upstream/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```
Also affects 2.3.0.  Testing results the same
```
ansible 2.3.0.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### CONFIGURATION
ANSIBLE_NOCOWS=1

ansible.cfg:
```
[persistent_connection]
connect_timeout = 30
connect_retries = 30
connect_interval = 1
```

##### OS / ENVIRONMENT
Linux (Debian and RHEL distributions)

##### SUMMARY
Missing information from registered variable when files changed when running in --check mode. Primarily missing invocation information, but also missing checksum, dest, gid, group, mode, owner, path, size, state, and uid.  Some of those may only be available if the file is changed for real, but invocation seems like it should be present regardless of check mode.  
All the information is present in --check mode if the file remains unchanged.

##### STEPS TO REPRODUCE
```yaml
---
- name: Test template output
  hosts: localhost
  vars:
    test_files:
      - unchanged_file
      - changed_file
      - new_file

  tasks:
    - name: Install files
      template:
        src: "{{ test_file }}"
        dest: "/tmp/{{ test_file }}"
      with_items: "{{ test_files }}"
      loop_control:
        loop_var: test_file
      register: install_files_task

    - name: registered var contents
      debug:
        var: install_files_task
```

##### EXPECTED RESULTS
```
TASK [registered var contents] ************************************************************************************************************************************************************
ok: [localhost] => {
    "install_files_task": {
        "changed": true, 
        "msg": "All items completed", 
        "results": [
            {
                "_ansible_item_label": "unchanged_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": false, 
                "diff": {
                    "after": {
                        "path": "/tmp/unchanged_file"
                    }, 
                    "before": {
                        "path": "/tmp/unchanged_file"
                    }
                }, 
                "gid": 1000, 
                "group": "szynaka", 
                "invocation": {
                    "module_args": {
                        "attributes": null, 
                        "backup": null, 
                        "content": null, 
                        "delimiter": null, 
                        "dest": "/tmp/unchanged_file", 
                        "diff_peek": null, 
                        "directory_mode": null, 
                        "follow": true, 
                        "force": false, 
                        "group": null, 
                        "mode": null, 
                        "original_basename": "unchanged_file", 
                        "owner": null, 
                        "path": "/tmp/unchanged_file", 
                        "recurse": false, 
                        "regexp": null, 
                        "remote_src": null, 
                        "selevel": null, 
                        "serole": null, 
                        "setype": null, 
                        "seuser": null, 
                        "src": null, 
                        "state": null, 
                        "unsafe_writes": null, 
                        "validate": null
                    }
                }, 
                "mode": "0664", 
                "owner": "szynaka", 
                "path": "/tmp/unchanged_file", 
                "size": 24, 
                "state": "file", 
                "test_file": "unchanged_file", 
                "uid": 1000
            }, 
            {
                "_ansible_item_label": "changed_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": true, 
                "checksum": "23086a8dbd5fd30b4483edc6107cf250bee87da9", 
                "dest": "/tmp/changed_file", 
                "diff": {
                    "after": "changed_file new content\n", 
                    "after_header": "dynamically generated", 
                    "before": "changed_file old content\n", 
                    "before_header": "/tmp/changed_file"
                }, 
                "gid": 1000, 
                "group": "szynaka", 
                "invocation": {
                    "module_args": {
                        "attributes": null, 
                        "backup": false, 
                        "content": null, 
                        "delimiter": null, 
                        "dest": "/tmp/changed_file", 
                        "directory_mode": null, 
                        "follow": true, 
                        "force": true, 
                        "group": null, 
                        "mode": null, 
                        "original_basename": "changed_file", 
                        "owner": null, 
                        "regexp": null, 
                        "remote_src": null, 
                        "selevel": null, 
                        "serole": null, 
                        "setype": null, 
                        "seuser": null, 
                        "src": "/home/szynaka/.ansible/tmp/ansible-tmp-1494878076.07-29070802577642/source", 
                        "unsafe_writes": null, 
                        "validate": null
                    }
                }, 
                "md5sum": "6ea8b6a698d2d4ec29e3aee03cc0bd41", 
                "mode": "0664", 
                "owner": "szynaka", 
                "size": 25, 
                "src": "/home/szynaka/.ansible/tmp/ansible-tmp-1494878076.07-29070802577642/source", 
                "state": "file", 
                "test_file": "changed_file", 
                "uid": 1000
            }, 
            {
                "_ansible_item_label": "new_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": true, 
                "checksum": "4054b2d417abb9cee4bf2177c081f7647885ce15", 
                "dest": "/tmp/new_file", 
                "diff": {
                    "after": "new_file contents\n", 
                    "after_header": "dynamically generated", 
                    "before": ""
                }, 
                "gid": 1000, 
                "group": "szynaka", 
                "invocation": {
                    "module_args": {
                        "attributes": null, 
                        "backup": false, 
                        "content": null, 
                        "delimiter": null, 
                        "dest": "/tmp/new_file", 
                        "directory_mode": null, 
                        "follow": true, 
                        "force": true, 
                        "group": null, 
                        "mode": null, 
                        "original_basename": "new_file", 
                        "owner": null, 
                        "regexp": null, 
                        "remote_src": null, 
                        "selevel": null, 
                        "serole": null, 
                        "setype": null, 
                        "seuser": null, 
                        "src": "/home/szynaka/.ansible/tmp/ansible-tmp-1494878076.66-268524017400749/source", 
                        "unsafe_writes": null, 
                        "validate": null
                    }
                }, 
                "md5sum": "bfc72d55e729a2d381b5988464d2d76e", 
                "mode": "0664", 
                "owner": "szynaka", 
                "size": 18, 
                "src": "/home/szynaka/.ansible/tmp/ansible-tmp-1494878076.66-268524017400749/source", 
                "state": "file", 
                "test_file": "new_file", 
                "uid": 1000
            }
        ]
    }
}
```

##### ACTUAL RESULTS

```
$ ansible-playbook template.yml --diff --check -vvvv
Using /etc/ansible/ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available

Loading callback plugin default of type stdout, v2.0 from /home/szynaka/src/ansible_upstream/lib/ansible/plugins/callback/__init__.pyc

PLAYBOOK: template.yml ********************************************************************************************************************************************************************
1 plays in template.yml

PLAY [Test template output] ***************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/basic.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/facts.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/_text.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/pycompat24.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/__init__.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/_six.py
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: szynaka
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024 `" && echo ansible-tmp-1494878043.52-21839276500024="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpllbX8l TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024/setup.py; rm -rf "/home/szynaka/.ansible/tmp/ansible-tmp-1494878043.52-21839276500024/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [Install files] **********************************************************************************************************************************************************************
task path: /home/szynaka/tmp/temlate_test/template.yml:11
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/_text.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/basic.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/pycompat24.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/__init__.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/_six.py
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: szynaka
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070 `" && echo ansible-tmp-1494878044.91-19445694350070="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpCfMAnS TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070/stat.py; rm -rf "/home/szynaka/.ansible/tmp/ansible-tmp-1494878044.91-19445694350070/" > /dev/null 2>&1 && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649 `" && echo ansible-tmp-1494878045.01-126034279435649="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649 `" ) && sleep 0'
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> PUT /tmp/tmpFXg27J TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/stat.py && sleep 0'
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/_text.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/basic.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/pycompat24.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/__init__.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/_six.py
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/file.py
<127.0.0.1> PUT /tmp/tmp34U0_U TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/file.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/file.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/file.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.01-126034279435649/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => (item=unchanged_file) => {
    "changed": false, 
    "diff": {
        "after": {
            "path": "/tmp/unchanged_file"
        }, 
        "before": {
            "path": "/tmp/unchanged_file"
        }
    }, 
    "gid": 1000, 
    "group": "szynaka", 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": null, 
            "content": null, 
            "delimiter": null, 
            "dest": "/tmp/unchanged_file", 
            "diff_peek": null, 
            "directory_mode": null, 
            "follow": true, 
            "force": false, 
            "group": null, 
            "mode": null, 
            "original_basename": "unchanged_file", 
            "owner": null, 
            "path": "/tmp/unchanged_file", 
            "recurse": false, 
            "regexp": null, 
            "remote_src": null, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": null, 
            "state": null, 
            "unsafe_writes": null, 
            "validate": null
        }
    }, 
    "mode": "0664", 
    "owner": "szynaka", 
    "path": "/tmp/unchanged_file", 
    "size": 24, 
    "state": "file", 
    "test_file": "unchanged_file", 
    "uid": 1000
}
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614 `" && echo ansible-tmp-1494878045.26-6193902917614="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpHgm8pj TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614/stat.py; rm -rf "/home/szynaka/.ansible/tmp/ansible-tmp-1494878045.26-6193902917614/" > /dev/null 2>&1 && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829 `" && echo ansible-tmp-1494878045.35-256228774704829="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829 `" ) && sleep 0'
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> PUT /tmp/tmpm9l6sV TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829/stat.py && sleep 0'
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/file.py
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419 `" && echo ansible-tmp-1494878045.45-131225397474419="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpE8u5kk TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419/file.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419/file.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.45-131225397474419/file.py && sleep 0'
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/basic.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/_text.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/pycompat24.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/__init__.py
Using module_utils file /home/szynaka/src/ansible_upstream/lib/ansible/module_utils/six/_six.py
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/net_tools/basics/slurp.py
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572 `" && echo ansible-tmp-1494878045.59-275468619018572="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmp8NwkvP TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572/slurp.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572/slurp.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.59-275468619018572/slurp.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.35-256228774704829/ > /dev/null 2>&1 && sleep 0'
--- before: /tmp/changed_file
+++ after: dynamically generated
@@ -1 +1 @@
-changed_file old content
+changed_file new content

changed: [localhost] => (item=changed_file) => {
    "changed": true, 
    "diff": {
        "after": "changed_file new content\n", 
        "after_header": "dynamically generated", 
        "before": "changed_file old content\n", 
        "before_header": "/tmp/changed_file"
    }, 
    "test_file": "changed_file"
}
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130 `" && echo ansible-tmp-1494878045.68-152701263031130="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpue3V21 TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130/stat.py; rm -rf "/home/szynaka/.ansible/tmp/ansible-tmp-1494878045.68-152701263031130/" > /dev/null 2>&1 && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963 `" && echo ansible-tmp-1494878045.77-151986799962963="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963 `" ) && sleep 0'
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/stat.py
<127.0.0.1> PUT /tmp/tmp9H3RZa TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963/stat.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963/stat.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963/stat.py && sleep 0'
Using module file /home/szynaka/src/ansible_upstream/lib/ansible/modules/files/file.py
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978 `" && echo ansible-tmp-1494878045.86-103252812205978="` echo /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmp1Z_XIQ TO /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978/file.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978/ /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978/file.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.86-103252812205978/file.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/szynaka/.ansible/tmp/ansible-tmp-1494878045.77-151986799962963/ > /dev/null 2>&1 && sleep 0'
--- before
+++ after: dynamically generated
@@ -0,0 +1 @@
+new_file contents

changed: [localhost] => (item=new_file) => {
    "changed": true, 
    "diff": {
        "after": "new_file contents\n", 
        "after_header": "dynamically generated", 
        "before": ""
    }, 
    "test_file": "new_file"
}

TASK [registered var contents] ************************************************************************************************************************************************************
task path: /home/szynaka/tmp/temlate_test/template.yml:20
ok: [localhost] => {
    "install_files_task": {
        "changed": true, 
        "msg": "All items completed", 
        "results": [
            {
                "_ansible_item_label": "unchanged_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "_ansible_parsed": true, 
                "changed": false, 
                "diff": {
                    "after": {
                        "path": "/tmp/unchanged_file"
                    }, 
                    "before": {
                        "path": "/tmp/unchanged_file"
                    }
                }, 
                "gid": 1000, 
                "group": "szynaka", 
                "invocation": {
                    "module_args": {
                        "attributes": null, 
                        "backup": null, 
                        "content": null, 
                        "delimiter": null, 
                        "dest": "/tmp/unchanged_file", 
                        "diff_peek": null, 
                        "directory_mode": null, 
                        "follow": true, 
                        "force": false, 
                        "group": null, 
                        "mode": null, 
                        "original_basename": "unchanged_file", 
                        "owner": null, 
                        "path": "/tmp/unchanged_file", 
                        "recurse": false, 
                        "regexp": null, 
                        "remote_src": null, 
                        "selevel": null, 
                        "serole": null, 
                        "setype": null, 
                        "seuser": null, 
                        "src": null, 
                        "state": null, 
                        "unsafe_writes": null, 
                        "validate": null
                    }
                }, 
                "mode": "0664", 
                "owner": "szynaka", 
                "path": "/tmp/unchanged_file", 
                "size": 24, 
                "state": "file", 
                "test_file": "unchanged_file", 
                "uid": 1000
            }, 
            {
                "_ansible_item_label": "changed_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "changed": true, 
                "diff": {
                    "after": "changed_file new content\n", 
                    "after_header": "dynamically generated", 
                    "before": "changed_file old content\n", 
                    "before_header": "/tmp/changed_file"
                }, 
                "test_file": "changed_file"
            }, 
            {
                "_ansible_item_label": "new_file", 
                "_ansible_item_result": true, 
                "_ansible_no_log": false, 
                "changed": true, 
                "diff": {
                    "after": "new_file contents\n", 
                    "after_header": "dynamically generated", 
                    "before": ""
                }, 
                "test_file": "new_file"
            }
        ]
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP ********************************************************************************************************************************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0 
```

### PR DESCRIPTION
**Pull request by chronidev**
##### SUMMARY
"Fixes #24633"

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
copy

### CODE REVIEW

**Comment by chronidev** [comment]
This affect 2.7 too.  2.6 2.5 2.4 2.3 2.2 are also affected but need this PR to be backported  https://github.com/ansible/ansible/pull/41426 previously to apply this one.

File: lib/ansible/plugins/action/copy.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def _ensure_invocation(self, result):
             if self._play_context.no_log:
                 result['invocation'] = "CENSORED: no_log is set"
             else:
-                result["invocation"] = self._task.args.copy()
```
Comment (jborean93) [inline lib/ansible/plugins/action/copy.py:219]: I would keep this line so we preserve the existing behaviour without breaking people's playbooks. Best to add a comment saying we know this is wrong but need it to keep backwards compatibility.

File: lib/ansible/plugins/action/copy.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ def _ensure_invocation(self, result):
...
+                result["invocation"] = dict(module_args=self._task.args.copy())
```
Comment (jborean93) [inline lib/ansible/plugins/action/copy.py]: Now that the above will be changed, you can do `result["invocation"]["module_args"] = self._task.args.copy()`

Comment (chronidev) [reply lib/ansible/plugins/action/copy.py]: @jborean93 No I can't, result["invocation"] doesn't exist at this time.

Comment (jborean93) [reply lib/ansible/plugins/action/copy.py]: It will if you re-add the line above, see https://github.com/ansible/ansible/pull/51582#discussion_r252909279

Comment (chronidev) [reply lib/ansible/plugins/action/copy.py]: Ok I understood. Let me do the change.

Comment (chronidev) [reply lib/ansible/plugins/action/copy.py]: done

**Review by jborean93 — APPROVED** [review]

**Pull request merged**